plugins/csp-vr-accessibility/src/FovVignette.cpp

```diff
@@ -0,0 +1,313 @@
+
+bool FovVignette::GetBoundingBox(VistaBoundingBox& bb) {
+  return false;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+float FovVignette::getNewRadius(
+    float innerOuterRadius, float normVelocity, float lastRadius, double dT) {
+  /* Experimental Code for Framerate Compensation
+  // target change from last radius (in %) per target time delta (avg frame time)
+  float targetDiff = 0.1F;
+  auto targetDtime = static_cast<float>(GetVistaSystem()->GetFrameLoop()->GetAverageLoopTime());
+  float diffPerTargetDt = targetDiff / targetDtime;
+
+  // (dT / targetDtime) >1 if dT > targetDtime, <1 if dT < targetDtime
+  float x = diffPerTargetDt * (static_cast<float>(dT) / targetDtime);
+  */
+
+  // targetRadius based on interpolation between maxRadius and innerOuterRadius
+  float targetRadius = ((1 - normVelocity) * sqrtf(2)) + (normVelocity * innerOuterRadius);
+
+  // newRadius increased towards targetRadius but limited to ~targetDiff changes
+  return (0.99F * lastRadius) + (0.01F * targetRadius);
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void FovVignette::updateDynamicRadiusVignette() {
+  // get simulation variables
+  float velocity = mSolarSystem->pCurrentObserverSpeed.get() /
+                   static_cast<float>(mSolarSystem->getObserver().getAnchorScale());
+  auto now = std::chrono::high_resolution_clock::now();
+
+  mNormalizedVelocity = (velocity - mVignetteSettings->mFovVignetteLowerVelocityThreshold.get()) /
+                        (mVignetteSettings->mFovVignetteUpperVelocityThreshold.get() -
+                            mVignetteSettings->mFovVignetteLowerVelocityThreshold.get());
+  auto deltaTime = static_cast<double>(
+      std::chrono::duration_cast<std::chrono::nanoseconds>(now - mLastTime).count());
+  mLastTime = now;
+
+  // clamp NormalizedVelocity to upper threshold
+  mNormalizedVelocity = (mNormalizedVelocity > 1.0F) ? 1.0F : mNormalizedVelocity;
+
+  mCurrentInnerRadius = getNewRadius(mVignetteSettings->mFovVignetteInnerRadius.get(),
+      mNormalizedVelocity, mLastInnerRadius, deltaTime);
+  mCurrentOuterRadius = getNewRadius(mVignetteSettings->mFovVignetteOuterRadius.get(),
+      mNormalizedVelocity, mLastOuterRadius, deltaTime);
+
+  // update variables
+  mLastInnerRadius = mCurrentInnerRadius;
+  mLastOuterRadius = mCurrentOuterRadius;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void FovVignette::updateFadeAnimatedVignette() {
+  // get simulation variables
+  float velocity = mSolarSystem->pCurrentObserverSpeed.get() /
+                   static_cast<float>(mSolarSystem->getObserver().getAnchorScale());
+  double currentTime =
+      cs::utils::convert::time::toSpice(boost::posix_time::microsec_clock::universal_time());
```

Comment:
What is the reason behind using spice time and the boost clock, instead of `std::chrono`?

Reply:
postponed for later additions to the plugin